spatialdata/_core/_spatialdata_ops.py

```diff
@@ -291,154 +287,76 @@ def _describe_paths(paths: list[list[Union[int, str]]]) -> str:
-            if not (
-                isinstance(region, str)
-                and region_key is None
-                or isinstance(region, list)
-                and isinstance(region_key, str)
-            ):
-                # this code should be never reached because the validate function should have raised an exception, but
-                # let's be extra safe
-                raise RuntimeError("Tables have incompatible region keys")
-            if isinstance(region, list):
-                all_region_keys.append(region_key)
-            else:
-                assert (
-                    len(all_region_keys) == 0
-                    or len(set(all_region_keys)) == 1
-                    and all_region_keys[0] == MERGED_TABLES_REGION_KEY
-                )
-                table.obs[MERGED_TABLES_REGION_KEY] = region
-                all_region_keys.append(MERGED_TABLES_REGION_KEY)
-        if len(set(all_region_keys)) != 1:
-            raise RuntimeError("Tables have incompatible region keys")
-        merged_region_key = all_region_keys[0]
-
-        all_regions = []
-        for table in tables:
-            region = table.uns[TableModel.ATTRS_KEY][TableModel.REGION_KEY]
-            if isinstance(region, str):
-                all_regions.append(region)
-            else:
-                all_regions.extend(region)
-        all_regions = list(set(all_regions))
-        merged_region = all_regions
-
-    attr = {"region": merged_region, "region_key": merged_region_key, "instance_key": merged_instance_key}
-    merged_table = anndata.concat(tables, join="outer", uns_merge="same")
-
-    # remove the MERGED_TABLES_REGION_KEY column if it has been added (the code above either adds that column
-    # to all the tables, either it doesn't add it at all)
-    for table in tables:
-        if MERGED_TABLES_REGION_KEY in table.obs:
-            del table.obs[MERGED_TABLES_REGION_KEY]
-
-    merged_table.uns[TableModel.ATTRS_KEY] = attr
-    merged_table.obs[merged_region_key] = merged_table.obs[merged_region_key].astype("category")
-    TableModel().validate(merged_table)
-    return merged_table
-
-
-def concatenate(sdatas: list[SpatialData], omit_table: bool = False) -> SpatialData:
-    """Concatenate a list of spatial data objects.
+            tables_l.append(table)
+            regions_l.append(regs)
+            instance_keys_l.append(table.obs[i].values.flatten())
+
+    merged_table = anndata.concat(tables_l, label=region_key, keys=regions_l, **kwargs)
+    merged_table.obs[instance_key] = np.concatenate(instance_keys_l, axis=0)
+    attrs = {
+        TableModel.REGION_KEY: regions_l,
+        TableModel.REGION_KEY_KEY: region_key,
+        TableModel.INSTANCE_KEY: instance_key,
+    }
+    merged_table.uns[TableModel.ATTRS_KEY] = attrs
+
+    return TableModel().validate(merged_table)
+
+
+def concatenate(
+    sdatas: list[SpatialData],
+    region_key: str = "region_merged",
```

Comment:
If all the `region_key` across the tables are identical, (`len(region_keys` == 1) I would use this value. So I would define `region_key` as `region_key: Optional[str] = None`.